source/src/protocols/motifs/LigandDiscoverySearch.cc

```diff
@@ -292,694 +366,1790 @@ utility::vector1<core::conformation::ResidueOP> LigandDiscoverySearch::get_all_r
+	working_pose_ = original_poseop;
+}
+
+//function to get a sub-library of motifs from the main library, based on the residue being used (only get for select residue)
+//function may have use outside discover, so public can use it
+protocols::motifs::MotifCOPs LigandDiscoverySearch::get_motif_sublibrary_by_aa(std::string residue_name)
+{
+	static basic::Tracer ms_tr( "LigandDiscoverySearch.get_motif_sublibrary_by_aa", basic::t_info );
+
+	//create temporary motifcops object to hold
+	protocols::motifs::MotifCOPs motif_holder;
+
+	for ( auto motifcop : motif_library_ ) {
+		//collect residue name from motifcop
+		std::string motif_residue_name(motifcop->restype_name1());
+
+		//if residue name matches that of the residue at the working position, add it to the motif_holder
+		if ( motif_residue_name == residue_name ) {
+			motif_holder.push_back(motifcop);
+		}
+	}
+
+	//assign motif_holder contents to motif_library_for_select_residue_
+	motif_library_for_select_residue_ = motif_holder;
+
+
+	ms_tr.Debug << "Created motif sub-library for residue " << residue_name << " with " << motif_library_for_select_residue_.size() << " motifs in it." << std::endl;
+
+
+	return motif_holder;
+}
+
+//create protein_representation_matrix_
+//uses working_pose to make the matrix
+void LigandDiscoverySearch::create_protein_representation_matrix(core::Size & x_shift, core::Size & y_shift, core::Size & z_shift, int & x_bound_int, int & y_bound_int, int & z_bound_int)
+{
+
+	//create tracer to identify points of the run
+	static basic::Tracer ms_tr( "LigandDiscoverySearch.create_protein_representation_matrix", basic::t_info );
+
+	//run through all atoms to derive a range of dimensions to contain the protein in a 3D  space
+	//since we can't have negative indices, we need to normalize the coordinate values so that everything is positive
+	//derive constant values based  on the most negative values in each dimension, and then add that constant to all coordinates
+
+	int smallest_x = 1;
+	int smallest_y = 1;
+	int smallest_z = 1;
+
+	int largest_x = 1;
+	int largest_y = 1;
+	int largest_z = 1;
+
+	//create a list of coordinates of each atom to hold and work with to fill the protein_representation_matrix
+	//can't seem to make a vector of xyzVector objects, so will need to just make a custome 2D vector  to  hold the data
+	utility::vector1<numeric::xyzVector<int>> atom_coordinates;
+
+	//determine largest and smallest x,y,z  values to determine dimensions of matrix
+	for ( core::Size res_num = 1; res_num <= working_pose_->size(); ++res_num ) {
+		for ( core::Size atom_num = 1; atom_num <= working_pose_->residue(res_num).natoms(); ++atom_num ) {
+			//get the x,y,z data of the atom
+			numeric::xyzVector<int> atom_xyz = working_pose_->residue(res_num).xyz(atom_num);
+			//convert the coordinates to integers
+			atom_xyz.x() = static_cast<int>(atom_xyz.x());
+			atom_xyz.y() = static_cast<int>(atom_xyz.y());
+			atom_xyz.z() = static_cast<int>(atom_xyz.z());
```

Comment:
You've already cast to integers in the line above. atom_xyz can't hold anything except integers, so all the values need to be converted during assignment.

Reply:
BTW, hopefully you're aware of the rules that C++ uses to truncate Real numbers to integers, and you're okay with the behavior on both the positive and negative side of zero.